lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -649,4 +681,159 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
+            if (!objidExists) {
+                return StatusEvent.
+                        getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(),
+                                StatusEventType.DELETE_ALL_VERSIONS).
+                        withNullableAccessGroupID(wsid).
+                        withNullableObjectID(Long.toString(objid)).build();
+            }
+        } catch (IOException ex) {
+            throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage());
+        } catch (JsonClientException ex) {
+            handleException(ex);
+        }
+
+        return ev;
+    }
+
+    private String getLatestName(StatusEvent ev)
+                     throws IndexingException,
+                            RetriableIndexingException {
+        String latestName = null;
+
+        if (ev.getNewName().isPresent()) {
+            latestName = ev.getNewName().get().toString();
+        }
+
+        Map<String, Object> command;
+        // wsid and objid of the object that the specified StatusEvent is an event for
+        final long wsid = ev.getAccessGroupId().get();
+        final long objid = Long.decode(ev.getAccessGroupObjectId().get());
+
+        // check if name has changed and update state of lastestName
+        try {
+            final String name = getObjectInfo(wsid, objid, Optional.absent()).
+                                                              getInfos().get(0).getE2();
+
+            latestName = name;
+
+        } catch (IOException e) {
+            throw handleException(e);
+        } catch (JsonClientException e) {
+            throw handleException(e);
+        }
+
+        return latestName;
+    }
+
+    private Boolean getLatestIsPublic(StatusEvent ev)
+                            throws IndexingException,
+                                   RetriableIndexingException {
+
+        Boolean latestIsPublic = null;
+        Map<String, Object> command;
+
+        if (ev.isPublic().isPresent()) {
+            latestIsPublic = ev.isPublic().get().booleanValue();
+        }
+
+        try {
+
+            final String isPublic = getWorkspaceInfo(ev.getAccessGroupId().get()).getE6();
+
+            latestIsPublic = (isPublic == "n") ? false: true;
+        } catch (IOException ex) {
+            throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage());
+        } catch (JsonClientException ex) {
+            handleException(ex);
+        }
+
+        return latestIsPublic;
+    }
+
+    @Override
+    public StatusEvent updateEvent(final StatusEvent ev)
+            throws IndexingException,
+                   RetriableIndexingException {
+
+        // only object level events are updated below
+        if (!ObjectEventQueue.isObjectLevelEvent(ev)) {
+            return ev;
+        }
+
+        // brute force get latest state and update event as event can be played
+        // out of order in the case of failed events being replayed.
+
+        // check deletion state
+        {
```

Comment:
Out of curiosity why the block?

Reply:
A form of encapsulation. The encapsulation helps me use and destroy the var updateEvent where I need it.  If I don't encapsulate updateEvent, I'd have to either reuse the var below where it is used a second time or name the two uses of the vars different like updateEvent1, updateEvent2, both of which I don't like.